### ISSUE TITLE
Read through Backprop as Functor, report how it fits into hasktorch

### ISSUE THREAD

**Issue by stites** [hasktorch/hasktorch#81]
I'm imagining that "Normalization layer" can be expressed in a neural network as some permutation of the following api change:

```haskell
data ConvLayer h w c = ConvLayer
  { conv :: Tensor '[h, w, c]
  }

-- to the following pseudo-haskell:
data ConvLayer h w c = ConvLayer
  { conv :: Normalize (Batch (Tensor '[h, w, c]))
  }
```
This would be akin to taking current deep learning patterns and codifying them into monoidal categories.

**Comment by stites** [comment]
From [Backprop as Functor](https://arxiv.org/pdf/1711.10455.pdf):
> In this paper we will show that learning becomes compositional — i.e. we can define a learning algorithm `A → C` from learning algorithms `A → B` and `B → C` — as long as each learning algorithm consists of these four components:
> - a parameter space `P`,
> - a function `I : P × A → B`,
> - an update function `U : P × A × B → P`, and
> - a request function `r : P × A × B → A`
>
> [...]
>
> Namely, we will show that `Learn` has the structure of a symmetric monoidal category.

**Comment by stites** [comment]
closing for the time being
